### PR TITLE
Transpile in loose mode and specify correct babel runtime

### PR DESCRIPTION
**Pull request by TrySound**
Loose mode works well here. Small size improvement.

About babel runtime. It should be in sync with others versions because
with every next version it may be changed. You can't be sure even these
versions do not cause bugs just because it works right now.

Syncing version with mui doesn't make sense. Then mui will be updated
and won't be in sync with this project. It's better to update whenever
possible to the latest versions.

There are different problems which we may solve. Do not stuck on helpers
please. You won't win a lot here. They will be 100% fixed when babel 7 will be
relased.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "presets": [
-    ["@babel/env", { "modules": false }],
-    "@babel/stage-3",
+    ["@babel/env", { "loose": true, "modules": false }],
```
Comment (oliviertassinari) [inline .babelrc]: Humm, I have tried that on Material-UI side, it was making the browser tests going crazy.

Comment (TrySound) [reply .babelrc]: I saw mui behaviour for the first time. In other projects it worked pretty well. And here it works good,

Comment (oliviertassinari) [reply .babelrc]: How can we know without a test suite?

**Comment by TrySound** [comment]
Did you checked it?

File: .babelrc (hunk 1 above)
Comment (oliviertassinari) [reply .babelrc]: Oh, unless it was a phantomJS issue only

Comment (oliviertassinari) [reply .babelrc]: Ok, I think that we should wait to enable the loose mode on Material-UI first before trying to rely on the loose mode.

Comment (TrySound) [reply .babelrc]: Probably.

It's safe in a dozens of projects I maintain. Mui problem may be to different stuff.

Comment (oliviertassinari) [reply .babelrc]: There is also one conceptual issue with the loose mode. The implementation doesn't match the specification. You might very well endup in a case where you rely on some edge case of the transpilation that will no longer work once you remove Babel to use the native feature.

Comment (oliviertassinari) [reply .babelrc]: By the way, we can remove PhantomJS and replace it with ChromeHeadless, puppeteer ease the installation overhead.

Comment (TrySound) [reply .babelrc]: That's why you should test the code without transpilation.

**Comment by TrySound** [comment]
Oliver, stop breaking babel without discussions.

**Comment by oliviertassinari** [comment]
What do you mean by breaking babel?

**Comment by TrySound** [comment]
@oliviertassinari Read the topic please. babel runtime is upgraded for the reason.

**Comment by oliviertassinari** [comment]
Oh. I see the problem. Ok, let's go back to babel.42.

**Comment by oliviertassinari** [comment]
https://github.com/zeit/next.js/blob/54e152b11bedea0c88e1799d27ed815160b7b040/package.json#L61

**Comment by TrySound** [comment]
47

**Comment by TrySound** [comment]
Who cares next js?

**Comment by oliviertassinari** [comment]
https://github.com/mui-org/material-ui/blob/2e7da5b68b1f9b4ba178b4338e4913f34bcc8383/package.json#L63

**Comment by oliviertassinari** [comment]
I do care about Next. We use it for the documentation and to power our product at the office.

**Comment by oliviertassinari** [comment]
Better, I think that we can try to keep this entropy to stress test it. See if people report issues about ti.

**Comment by TrySound** [comment]
Sticking to betas is a wasting of a time. As I said a few times. You won't win a lot here.
https://github.com/renatorib/react-powerplug/blob/master/package.json#L63

**Comment by TrySound** [comment]
Shooting in the feet. :(

**Comment by oliviertassinari** [comment]
One more thought. If the link between babel-runtime and babel-transform-runtime was **that important**, we would need to use a fixed version of babel-runtime, not a scoped version (^). For instance, It's not what `react-powerplug` is doing, it's taking the risk to break on the next release of babel-runtime. So, I would rather be greedy, try to maximize deduplicate, and do a step backward if some people raise an issue.

How does that sounds?

**Comment by TrySound** [comment]
package managers are already stick with prerelease versions so this is not a problem. (^) doesn't change anything.

Duplication is not a problem. We still win comparing to inlined helpers. Sticking to one version doesn't work. If the winner will be 47 all 42 helpers will be duplicated for all packages because the winner is hoisted and you can't do anything with this. Use the latest one and don't waste a time trying to sync these versions.

**Comment by oliviertassinari** [comment]
> package managers are already stick with prerelease versions so this is not a problem. (^) doesn't change anything.

Not on a new installation

> Use the latest one and don't waste a time trying to sync these versions.

Isn't what we do?

We use babel@7.0.0-beta.47 to transpile the source. Still, we are accepting older runtime.

**Comment by TrySound** [comment]
@oliviertassinari Well, you can break the build doing this. There was a breaking change before 42. They changed helpers. You are saving a few KB but your project may crash with the next dependency upgrade.

**Comment by TrySound** [comment]
You are right. Sticking is a good idea. That's my fault.

**Comment by TrySound** [comment]
But sticking to different version can break one day.

**Comment by oliviertassinari** [comment]
> sticking to different version can break one day.

I agree, it **will** break one day.
Following the same strategy, I try to keep the `^` scope as wide a possible on the `package.json` while always beeing up to date on the `yarn.lock`. I have found it to be a nice tradeoff between deduplication and "up-to-date-ness".

**Comment by oliviertassinari** [comment]
> Duplication is not a problem

You were right, the overhead is minimal, it's a simple proxy to core-js
capture d ecran 2018-05-21 a 18 52 18

**Comment by TrySound** [comment]
builtIns helpers don't use core-js

**Pull request merged**